hw/ip/otbn/util/check_loop.py

```diff
@@ -0,0 +1,250 @@
+#!/usr/bin/env python3
+# Copyright lowRISC contributors.
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+
+import argparse
+import sys
+from typing import Dict, List, Tuple
+
+from shared.check import CheckResult
+from shared.decode import OTBNProgram, decode_elf
+from shared.insn_yaml import Insn
+
+
+class CodeSection:
+    '''A continuous part of a program's code (represented as a PC range).
+
+    The code section is considered to include both the start and end PC.
+    '''
+    def __init__(self, start: int, end: int):
+        self.start = start
+        self.end = end
+
+    def __contains__(self, pc):
+        return self.start <= pc and pc <= self.end
+
+    def __repr__(self):
+        return '0x{:x}-0x{:x}'.format(self.start, self.end)
```

Comment:
Does it make sense to do `{:08x}` to get zero padding?

Reply:
In `otbn-objdump` there's no zero padding, so it's a bit easier to cross-check if this matches. I think we should change both if we change one. I'm happy with either way, although it's worth noting that OTBN programs are usually pretty small.

Reply:
Also, `otbn-objdump` doesn't really touch this bit of the output: the addresses come straight out of `riscv32-unknown-elf-objdump`. So I'd probably suggest not doing the padding.